Make `xpack.actions.rejectUnauthorized` setting work

mikecote opened the pull request:
Resolves https://github.com/elastic/kibana/issues/87047

In this PR, I'm doing a few changes to fix the problem. The general purpose of each commit is as follows:
1. 78b7e2e82ac56dcd8f0c1b6b092ced3317dd8933: I'm removing the TS type `ActionsConfigType` because it is a duplicate / subset of `ActionsConfig` that wasn't needed. This allowed me to add more functionality to the `configurationUtilities` in following commits.
2. 08ebe63570f93180d89689740eddbc06bb39d7ac: This commit fixes the problem by passing `configurationUtilities` down to the `getProxyAgents` function from every possible path. The `configurationUtilities` has a new function `isRejectUnauthorizedCertificatesEnabled `.
3. 1a17f8ded53bd6d51d44be16ef43e0f85d7644f5: Since passing `proxySettings` was happening at the same places that `configurationUtilities` was, I moved the access to `proxySettings` to `configurationUtilities`.
4. a84503dfa483dfa571cb6f816dabfbd54d96356a: I realized checking `error.isAxiosError` didn't always work (ex: when it mentions missing kbn-xsrf header) so I changed it to `error.code`.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99624/)
* Commit: 1a17f8ded53bd6d51d44be16ef43e0f85d7644f5
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99624/execution/node/335/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99624/","number":"99624","commit":"1a17f8ded53bd6d51d44be16ef43e0f85d7644f5"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99743/)
* Commit: c5f6eb7c97763486235f6c736f63d6a3ddac879c

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #99624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99624/) failed 1a17f8ded53bd6d51d44be16ef43e0f85d7644f5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99624/","number":"99624","commit":"1a17f8ded53bd6d51d44be16ef43e0f85d7644f5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99743/","number":"99743","commit":"c5f6eb7c97763486235f6c736f63d6a3ddac879c"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-alerting-services (Team:Alerting Services)

YulNaumenko reviewed (approved):
LGTM

pmuellr reviewed (approved):
LGTM

Seems like we could build a function test for this - create a new https server with a newly created self-signed cert (with a loooong lifetime), that just has one endpoint that we can test with webhook.  Then execute the webhook, with and without the rejectUnauthorized setting - should get one to work and the other to fail.  Obviously set it up to use the proxy we're testing with, in security_and_spaces . :-)

mikecote commented:
@pmuellr I went ahead and added a functional test in this commit (https://github.com/elastic/kibana/pull/88690/commits/af638a3e94133ce1630f151461bfe3cc812f25db). I made it re-use the dev certs so it's one less thing to expire 🙂 I added it to the `spaces_only` suite because the fix is for non-proxy scenarios (the other suite uses a proxy for everything). Let me know what you think of the approach? 🙏

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101253/)
* Commit: af638a3e94133ce1630f151461bfe3cc812f25db
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101253/execution/node/327/log)

### History
* :green_heart: [Build #99969](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99969/) succeeded a84503dfa483dfa571cb6f816dabfbd54d96356a
* :green_heart: [Build #99743](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99743/) succeeded c5f6eb7c97763486235f6c736f63d6a3ddac879c
* :broken_heart: [Build #99624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99624/) failed 1a17f8ded53bd6d51d44be16ef43e0f85d7644f5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99624/","number":"99624","commit":"1a17f8ded53bd6d51d44be16ef43e0f85d7644f5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99743/","number":"99743","commit":"c5f6eb7c97763486235f6c736f63d6a3ddac879c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99969/","number":"99969","commit":"a84503dfa483dfa571cb6f816dabfbd54d96356a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101253/","number":"101253","commit":"af638a3e94133ce1630f151461bfe3cc812f25db"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101256/)
* Commit: 0337bbdf2f22d85c925b192074ed04db133e9bd2
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101256/execution/node/319/log)

### History
* :broken_heart: [Build #101253](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101253/) failed af638a3e94133ce1630f151461bfe3cc812f25db
* :green_heart: [Build #99969](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99969/) succeeded a84503dfa483dfa571cb6f816dabfbd54d96356a
* :green_heart: [Build #99743](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99743/) succeeded c5f6eb7c97763486235f6c736f63d6a3ddac879c
* :broken_heart: [Build #99624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99624/) failed 1a17f8ded53bd6d51d44be16ef43e0f85d7644f5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99624/","number":"99624","commit":"1a17f8ded53bd6d51d44be16ef43e0f85d7644f5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99743/","number":"99743","commit":"c5f6eb7c97763486235f6c736f63d6a3ddac879c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99969/","number":"99969","commit":"a84503dfa483dfa571cb6f816dabfbd54d96356a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101253/","number":"101253","commit":"af638a3e94133ce1630f151461bfe3cc812f25db"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101256/","number":"101256","commit":"0337bbdf2f22d85c925b192074ed04db133e9bd2"}]}
PIPELINE-->

mikecote commented:
@elasticmachine merge upstream

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101314/)
* Commit: d6be5073ecf9dd581dbb4f16b2c5e6893517c08b
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101314/execution/node/507/log)
* [Execute kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101314/execution/node/727/log)

### History
* :broken_heart: [Build #101256](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101256/) failed 0337bbdf2f22d85c925b192074ed04db133e9bd2
* :broken_heart: [Build #101253](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101253/) failed af638a3e94133ce1630f151461bfe3cc812f25db
* :green_heart: [Build #99969](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99969/) succeeded a84503dfa483dfa571cb6f816dabfbd54d96356a
* :green_heart: [Build #99743](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99743/) succeeded c5f6eb7c97763486235f6c736f63d6a3ddac879c
* :broken_heart: [Build #99624](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99624/) failed 1a17f8ded53bd6d51d44be16ef43e0f85d7644f5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99624/","number":"99624","commit":"1a17f8ded53bd6d51d44be16ef43e0f85d7644f5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99743/","number":"99743","commit":"c5f6eb7c97763486235f6c736f63d6a3ddac879c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99969/","number":"99969","commit":"a84503dfa483dfa571cb6f816dabfbd54d96356a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101253/","number":"101253","commit":"af638a3e94133ce1630f151461bfe3cc812f25db"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101256/","number":"101256","commit":"0337bbdf2f22d85c925b192074ed04db133e9bd2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101314/","number":"101314","commit":"d6be5073ecf9dd581dbb4f16b2c5e6893517c08b"}]}
PIPELINE-->

mikecote commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101736/)
* Commit: b9414e30a21ff37cb6a5a994fbee15f526fb99d5

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #101314](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101314/) failed d6be5073ecf9dd581dbb4f16b2c5e6893517c08b
* :broken_heart: [Build #101256](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101256/) failed 0337bbdf2f22d85c925b192074ed04db133e9bd2
* :broken_heart: [Build #101253](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101253/) failed af638a3e94133ce1630f151461bfe3cc812f25db
* :green_heart: [Build #99969](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99969/) succeeded a84503dfa483dfa571cb6f816dabfbd54d96356a
* :green_heart: [Build #99743](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99743/) succeeded c5f6eb7c97763486235f6c736f63d6a3ddac879c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99743/","number":"99743","commit":"c5f6eb7c97763486235f6c736f63d6a3ddac879c"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99969/","number":"99969","commit":"a84503dfa483dfa571cb6f816dabfbd54d96356a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101253/","number":"101253","commit":"af638a3e94133ce1630f151461bfe3cc812f25db"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101256/","number":"101256","commit":"0337bbdf2f22d85c925b192074ed04db133e9bd2"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101314/","number":"101314","commit":"d6be5073ecf9dd581dbb4f16b2c5e6893517c08b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101736/","number":"101736","commit":"b9414e30a21ff37cb6a5a994fbee15f526fb99d5"}]}
PIPELINE-->
